src/Diagnostics.DataProviders/DataProviderLogDecorator.cs

```diff
@@ -0,0 +1,114 @@
+        {
+            _observerDataProvider = dataProvider;
+        }
+
+        public async Task<DataTable> ExecuteQuery(string query, string stampName, string requestId = null, string operationName = null)
+        {
+            return await MakeDependencyCall(requestId, _kustoDataProvider.ExecuteQuery(query, stampName, requestId, operationName));
+        }
+
+        public async Task<List<IDictionary<string, dynamic>>> GetAppDeployments(string subscriptionId, string resourceGroupName, string name)
+        {
+            return await MakeDependencyCall(null, _geomasterDataProvider.GetAppDeployments(subscriptionId, resourceGroupName, name));
+        }
+
+        public async Task<IDictionary<string, string>> GetAppSettings(string subscriptionId, string resourceGroupName, string name)
+        {
+            return await MakeDependencyCall(null, _geomasterDataProvider.GetAppSettings(subscriptionId, resourceGroupName, name));
+        }
+
+        public async Task<dynamic> GetResource(string wawsObserverUrl)
+        {
+            return await MakeDependencyCall(null, _observerDataProvider.GetResource(wawsObserverUrl));
+        }
+
+        public async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMap(string siteName)
+        {
+            return await MakeDependencyCall(null, _observerDataProvider.GetRuntimeSiteSlotMap(siteName));
+        }
+
+        public async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMap(string stampName, string siteName)
+        {
+            return await MakeDependencyCall(null, _observerDataProvider.GetRuntimeSiteSlotMap(stampName, siteName));
+        }
+
+        public async Task<dynamic> GetSite(string siteName)
+        {
+            return await MakeDependencyCall(null, _observerDataProvider.GetSite(siteName));
+        }
+
+        public async Task<dynamic> GetSite(string stampName, string siteName)
+        {
+            return await MakeDependencyCall(null, _observerDataProvider.GetSite(stampName, siteName));
+        }
+
+        public async Task<IDictionary<string, string[]>> GetStickySlotSettingNames(string subscriptionId, string resourceGroupName, string name)
+        {
+            return await MakeDependencyCall(null, _geomasterDataProvider.GetStickySlotSettingNames(subscriptionId, resourceGroupName, name));
+        }
+
+        public async Task<VnetValidationRespone> VerifyHostingEnvironmentVnet(string subscriptionId, string vnetResourceGroup, string vnetName, string vnetSubnetName, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return await MakeDependencyCall(null, _geomasterDataProvider.VerifyHostingEnvironmentVnet(subscriptionId, vnetResourceGroup, vnetName, vnetSubnetName, cancellationToken));
+        }
+
+        private async Task<T> MakeDependencyCall<T>(string requestId, Task<T> dataProviderTask, [CallerMemberName]string dataProviderOperation = "")
+        {
```

Comment:
Do you see a problem here that we will not be able to measure how individual kusto queries will perform? Basically `dataProviderOperation` this is going to be `ExecuteQuery` every time. I am not sure what a good solution to this is but it seems like we will want greater granularity in terms of how specific queries are doing.

Reply:
I thought about this. But just to make sure we're on the same page. Who would be the stakeholders who are interested in this data? Would it be us the platform developers or the developers creating detectors?